zeebe/broker/src/test/java/io/camunda/zeebe/broker/exporter/metrics/MetricsExporterTest.java

```diff
@@ -96,6 +124,67 @@ void shouldObserveJobLifetime() {
         .describedAs("Expected the correct job_life_time bucket to have counted the event");
   }
 
+  @Test
+  void shouldCleanupMetricsWhenContextIsClosed() throws Exception {
```

Comment:
:wrench: I'd suggest making this a generic test for the behavior of `ExporterContext` instead of a test specific to the `MetricsExporter`. Something like: 
1. Create a context
2. Set some metrics
3. Assert that metrics are set with partitionId, exporterId labels

And then a second one to assert that the metrics are removed when the context is closed.